### PR TITLE
Docs setup and first pass at API reference

### PR DESCRIPTION
**Pull request by jrbotros**
First pass at docs can be seen at http://orchestra.readthedocs.org/en/docs/!

In this PR:
- Add skeleton for sphinx docs for use with readthedocs.org
- Add docstrings to all task_lifecycle methods and set up autodoc
- Add doc building to test suite to catch egregious docs errors

TODO:
- continue to docstring important modules
- flesh out high-level pages in conjunction with the open-source site verbiage

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+man_pages = [
+    (master_doc, 'orchestra', u'Orchestra Documentation',
+     [author], 1)
+]
+
+# -- Options for Texinfo output -------------------------------------------
+
+texinfo_documents = [
+    (master_doc, 'Orchestra', u'Orchestra Documentation',
+     author, 'Orchestra', ('An open source tool to orchestrate teams of '
```
Comment (marcua) [inline docs/source/conf.py]: tool->system

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                           'experts as they complete complex projects with '
+                           'the help of automation..'),
```
Comment (marcua) [inline docs/source/conf.py]: ..->.

File: docs/source/motivation.rst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Our Motivation
+==============
+
```
Comment (marcua) [inline docs/source/motivation.rst:3]: text is updated

**Comment by marcua** [comment]
lgtm: merge after addressing comments.

File: docs/source/conf.py (hunk 1 above)
Comment (jrbotros) [inline docs/source/conf.py]: :white_check_mark:

File: docs/source/conf.py (hunk 2 above)
Comment (jrbotros) [inline docs/source/conf.py]: :white_check_mark:

File: docs/source/motivation.rst (hunk 3 above)
Comment (jrbotros) [inline docs/source/motivation.rst:3]: :white_check_mark:

**Pull request merged**